### PR TITLE
Fix manual ducking

### PR DESCRIPTION
**Pull request by daschuer**
Now the ducking gain is only applied when talkover is enabled. 

Before the manual mode has always applied the Ducking gain. This does not make much sense, because the user can also lower the deck gain to have the same effect. 

This Fixes a a regression since the introduction of the postfader effects in 2.1

### CODE REVIEW

File: src/engine/enginesidechaincompressor.h
Comment (ronso0) [inline src/engine/enginesidechaincompressor.h:44]: This could be inline like the other setXyz() functions above.

**Comment by ronso0** [comment]
Nice, thank you for the fix. I was about to propose that fix.
Works nicely and prevents lowered / muted master volume after start with hidden Mics.

Fixes https://bugs.launchpad.net/mixxx/+bug/1394968 and I can close #2006. Thx!

File: src/engine/enginemaster.cpp
Comment (uklotzde) [inline src/engine/enginemaster.cpp:475]: In debug log in real-time code??

**Review by uklotzde — CHANGES_REQUESTED** [review]

File: src/engine/enginemaster.cpp
Comment (daschuer) [reply src/engine/enginemaster.cpp:475]: Ups ...

**Comment by daschuer** [comment]
Fixes: 
https://bugs.launchpad.net/mixxx/+bug/1394968
https://bugs.launchpad.net/mixxx/+bug/1737113
https://bugs.launchpad.net/mixxx/+bug/1662536

**Comment by ronso0** [comment]
Just that I get our Merge policy straight, finally:
* if we squash and merge this (after a LGTM from a c++ pro) the commit message doesn't fully reflect the two separate fixes
* if I create a Merge Commit we have "silnce debug" commit, as well, which just removes something that was introduced in a previous commit and shouldn't end up in the final commit history, right?

So if I get that right the first two commits should be squashed, then we Merge Commit, right?

**Comment by Holzhaus** [comment]
I think we should either
a) just merge and keep the history, or
b) do an interactive rebase, cut-and-paste the `pick 8af17c2 Silence Debug` right after the line that introduced the debug statement and replace `pick` with `fixup`.

**Comment by daschuer** [comment]
The general rule applies here, no rebase during a review. That's easy and saved all parties from extra work.

I must admit that since the found issue effects the top most commit, I should have sqashed that.
But since no one cares after a while lets save us from the work and merge this.

**Pull request merged**
